### PR TITLE
Coverity 03

### PR DESCRIPTION
**Pull request by andrewbird**
Third slice of coverity fixes.

### CODE REVIEW

**Comment by andrewbird** [comment]
Seems I broke the cpu tests :-(

**Comment by andrewbird** [comment]
I dropped the CBW/CWDE patch until somebody smarter than me can figure out what's wrong with it!

File: src/dosext/mfs/lfn.c [hunk 1]
```diff
@@ -683,7 +683,7 @@ static int wildcard_delete(char *fpath, int drive)
 		free(dir);
 		return lfn_error(PATH_NOT_FOUND);
 	}
-	pattern = malloc(strlen(slash + 1) + 1);
+	pattern = strdup(slash + 1);
```
Comment (stsp) [inline src/dosext/mfs/lfn.c:686]: I don't think that's right.
`pattern` is likely overwritten
a line below.

File: src/plugin/debugger/dosdebug.c [hunk 2]
```diff
@@ -109,18 +109,23 @@ static int find_dosemu_pid(const char *tmpfile, int local)
...
-  sprintf(name, "/proc/%d/stat", pid);
-  if ((fd = open(name, O_RDONLY)) ==-1) return 0;
-  if (read(fd, buf, sizeof(buf)-1) <=0) {
+  snprintf(name, sizeof(name), "/proc/%d/stat", pid);
+
+  if ((fd = open(name, O_RDONLY)) == -1)
+    return 0;
+
+  num = read(fd, buf, sizeof(buf) - 1);
+  if (num == -1) {
```
Comment (stsp) [inline src/plugin/debugger/dosdebug.c:121]: Perhaps also 0 (EOF)?

**Comment by stsp** [comment]
So what was wrong with CBW et al?
Have you made sure they break tests?
Seems like you were uncertain.

File: src/dosext/mfs/lfn.c (hunk 1 above)
Comment (andrewbird) [reply src/dosext/mfs/lfn.c:686]: Yes, the copying of the string data was incidental to the allocation, but certainly it doesn't make things clearer. I'll think up something else!

Comment (andrewbird) [reply src/dosext/mfs/lfn.c:686]: Is this okay?
```diff
-       pattern = malloc(strlen(slash + 1) + 1);
-       name_ufs_to_dos(pattern, slash + 1);
+       pattern = malloc(strlen(&slash[1]) + 1);
+       name_ufs_to_dos(pattern, &slash[1]);
```

Comment (stsp) [reply src/dosext/mfs/lfn.c:686]: Maybe, but I am not really sure
what was the problem.
It looks like the same thing!

Comment (andrewbird) [reply src/dosext/mfs/lfn.c:686]: I don't think there was a problem except that Coverity views `malloc(strlen(string + 1))` as a mistake and that the dev probably meant `malloc(strlen(string) + 1)`. I'm quite happy to drop this,as it's not clearer with either of the two suggested changes, and too much effort just to please Coverity!

File: src/plugin/debugger/dosdebug.c (hunk 2 above)
Comment (andrewbird) [reply src/plugin/debugger/dosdebug.c:121]: Okay, will do.

**Comment by andrewbird** [comment]
> Have you made sure they break tests?
Seems like you were uncertain.

I never saw the cpu test fail randomly here, only on Travis. I've made the change you asked for, dropped the `malloc(strlen(` patch and re-added the CBW/CWDE patch. Let's see what happens on Travis.

**Comment by stsp** [comment]
Thanks!

**Pull request merged**